### PR TITLE
hydraulic_press

### PR DESCRIPTION
**Pull request by Darya2909**

### CODE REVIEW

File: symplyphysics/laws/hydro/hydraulic_press.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from sympy import (Eq, solve)
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hydraulic_press.py:1]: Don't use marketing names, like `Newton's first law` or `hydraulic press`. Use some descriptive name - how input parameters are linked to the result.

File: symplyphysics/laws/hydro/hydraulic_press.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+## S2 is the area of the second piston
+
+## Conditions
+## This ratio is performed only in an ideal hydraulic press, i.e. one in which there is no friction.
+
+
+first_force = Symbol("first_force", units.force)
+first_area = Symbol("first_area", units.area)
+second_force = Symbol("second_force", units.force)
+second_area = Symbol("second_area", units.area)
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hydraulic_press.py:24]: First and second are not very informative. How about `input` and `output`?

File: symplyphysics/laws/hydro/hydraulic_press.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+@validate_output(second_force)
+def calculate_force(first_force_: Quantity, first_area_, second_area_: Quantity) -> Quantity:
+    result_expr = solve(law, second_force, dict=True)[0][second_force]
+    result_force = result_expr.subs({
+        first_force: first_force_,
+        first_area: first_area_,
+        second_area: second_area_,
+    })
+
+    return Quantity(result_force)
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hydraulic_press.py:43]: Add empty line.

File: symplyphysics/laws/hydro/hydraulic_press.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+from sympy import (Eq, solve)
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+                           validate_output)
+
+# Description
+## If both vertically positioned cylinders of communicating vessels are closed with pistons, then with the help of external forces applied to the pistons,
+## a large pressure can be created in the liquid, many times exceeding the hydrostatic pressure at any point in the system.
+## If the pistons have different areas, then different forces act on them from the liquid side. The same modulus, but oppositely directed external forces must be applied to the pistons to keep the system in balance.
+
+## Law: F1 / S1 = F2 / S2
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hydraulic_press.py:10]: I'd like to see the proof as soon as this PR is merged.

File: test/hydro/hydraulic_press_test.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+from symplyphysics.laws.hydro import hydraulic_press as hydraulic
+
+
+@fixture(name="test_args")
+def test_args_fixture():
+    first_force = Quantity(2000 * units.newton)
+    first_area = Quantity(10 * units.meter**2)
+    second_area = Quantity(0.1 * units.meter ** 2)
```
Comment (blackyblack) [inline test/hydro/hydraulic_press_test.py:18]:
```suggestion
    second_area = Quantity(0.1 * units.meter**2)

```

File: test/hydro/hydraulic_press_test.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+
+def test_basic_force(test_args):
+    result = hydraulic.calculate_force(test_args.first_force, test_args.first_area, test_args.second_area)
+    assert SI.get_dimension_system().equivalent_dims(result.dimension, units.force)
+    result_force = convert_to(result, units.newton).evalf(5)
+    assert result_force == approx(20, 0.001)
+
+
+def test_bad_area_and_force(test_args):
```
Comment (blackyblack) [inline test/hydro/hydraulic_press_test.py:34]: Separate test cases.
You can combine together area tests or force tests but different entities should be separate.

File: symplyphysics/laws/hydro/hydraulic_press.py (hunk 4 above)
Comment (Darya2909) [reply symplyphysics/laws/hydro/hydraulic_press.py:10]: https://www.webmath.ru/poleznoe/fizika_127_gidravlicheskij_press.php

Comment (blackyblack) [reply symplyphysics/laws/hydro/hydraulic_press.py]: Yes, but I want to see it as a code.

File: symplyphysics/laws/hydro/piston_force_of_the_hydraulic_press.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+law = Eq(input_force / input_area, output_force / output_forces_area)
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(input_force_=input_force, input_area_=input_area, output_forces_area_=output_forces_area)
+@validate_output(output_force)
+def calculate_force(input_force_: Quantity, input_area_, output_forces_area_: Quantity) -> Quantity:
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/piston_force_of_the_hydraulic_press.py:35]:
```suggestion
def calculate_output_force(input_force_: Quantity, input_area_, output_forces_area_: Quantity) -> Quantity:
```

File: symplyphysics/laws/hydro/piston_force_of_the_hydraulic_press.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+from sympy import (Eq, solve)
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/piston_force_of_the_hydraulic_press.py:1]: Law file name is still not good. It is not about hydraulic press and it is not limited to hydraulic press applications.
I suggest `input_force_to_area_ratio_equals_to_output`

File: symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py [hunk 9]
```diff
@@ -14,6 +14,13 @@
 ## S1 is the area of the first piston
 ## S2 is the area of the second piston
 
+## Proof
+## The pressure that piston number one creates is equal to: p1 = F1 / S1
+## The pressure of the second piston on the liquid is: p2 = F2 / S2
+## If the pistons are in equilibrium, then the pressures p1 and p2 are equal, therefore, we can equate the right-hand sides of the expressions
+## ==> F1 / S1 = F2 / S2
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py:21]: Delete. As I told, I want to see it as code, not comments in the code. You can see an example of such proof here: https://github.com/blackyblack/symplyphysics/blob/a3df72abb95b37aada38e6929dfc1153b032f873/symplyphysics/laws/gravity/free_fall_acceleration_from_height.py#L27

File: symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py [hunk 10]
```diff
@@ -32,6 +28,21 @@
 
 law = Eq(input_force / input_area, output_force / output_forces_area)
 
+pressure_input = pressure_law.law.rhs.subs({
+    pressure_law.force: input_force,
+    pressure_law.area: input_area
+})
+
+## If the pistons are in equilibrium, then the pressures pressure_input and pressure_output are equal
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py:36]: To equate pressure on both sides, you should use this law: https://github.com/blackyblack/symplyphysics/blob/master/symplyphysics/laws/hydro/inner_pressure_of_fluid_is_constant.py
It is not very easy to use though. Here is an example, how to use it: https://github.com/blackyblack/symplyphysics/blob/ea325837f3684750a2fa1b4e68af4d3966a1960c/examples/hydro/pressure_balance_in_u_tube.py#L66

File: symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py [hunk 11]
```diff
@@ -46,5 +57,5 @@ def calculate_output_force(input_force_: Quantity, input_area_, output_forces_ar
         input_area: input_area_,
         output_forces_area: output_forces_area_,
     })
-
+    print(result_force)
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py:60]: Remove print.

File: symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    pressure_law.force: output_force,
+    pressure_law.area: output_forces_area,
+})
+
+## If the pistons are in equilibrium, then the pressures pressure_input and pressure_output are equal
+dsolved = dsolve(constant_pressure_law.law, constant_pressure_law.inner_pressure(constant_pressure_law.time))
+dsolved_input = dsolved.subs(constant_pressure_law.inner_pressure(constant_pressure_law.time), pressure_input)
+dsolved_output = dsolved.subs(constant_pressure_law.inner_pressure(constant_pressure_law.time), pressure_output)
+solved_input = solve([dsolved_input, dsolved_output], (pressure_input, "C1"), dict=True)[0][pressure_input]
+pressure_equation = Eq(pressure_input, solved_input)
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/input_force_to_area_ratio_equals_to_output.py:47]: You should verify that you've got the law, you're trying to prove. Use `assert` and `expr_equals`

**Review by blackyblack — APPROVED** [review]

**Pull request merged**